src/components/ProductList/useFiltering.js

```diff
@@ -0,0 +1,29 @@
+import {useState, useEffect} from 'react'
+
+const useFiltering = (data) => {
+    const [filters, setFilters] = useState([]);
+    const [filteredData, setFilteredData] = useState(data);
+
+    useEffect(() => {
+        if(filters.length > 0){
+            setFilteredData(data.filter(x => filters.includes(x.data.category.slug)));
```

Comment:
maybe you can find a better way to filter this and if you implement the hook for the data there you can parse the data and clean the data from the properties that are not useful. you can for example move the property slug to a top level so it's easier to access.

Reply:
Also instead of naming the property `x` in the filter try giving them a more significant name